ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -333,6 +343,7 @@ def advance(self, env: SubprocessEnvManager) -> int:
                 trainer.increment_step(len(new_step_infos))
                 if trainer.is_ready_update():
                     # Perform gradient descent with experience buffer
-                    trainer.update_policy()
+                    with hierarchical_timer("update_policy"):
```

Comment:
The decorator is more concise and doesn't need the whitespace change, but this is probably better for parts of the code where we're making calls to an abstract class.